src/Language/DifferentialDatalog/FlatBuffer.hs

```diff
@@ -385,22 +386,17 @@ typeTableName x =
                      | otherwise
                        -> "__BigInt"
          TTuple{..}    -> fbTupleName typeTupArgs
-         TUser{..} | typeHasUniqueConstructor x
+         TUser{..}   | typeHasUniqueConstructor x
                        -> fbStructName typeName typeArgs
-                   | otherwise
+                     | otherwise
                        -> "__Table_" <> fbStructName typeName typeArgs
-         TOpaque{typeArgs = [elemType], ..} | elem typeName sET_TYPES
-                       -> "__Table_Vec_" <> mkTypeIdentifier elemType
-         TOpaque{typeArgs = [keyType, valType], ..} | typeName == mAP_TYPE
-                       -> typeTableName $ tOpaque (mOD_STD ++ "::Vec") [tTuple [keyType, valType]]
+         t@TOpaque{} | typeIsIterable ?d t
+                       -> "__Table_Vec_" <> (mkTypeIdentifier $ fst $ typeIterType ?d t)
          t             -> error $ "typeTableName: Unexpected type " ++ show t
 
 -- True if type is serialized into a vector inside FlatBuffer.
 typeIsVector :: (WithType a, ?d::DatalogProgram) => a -> Bool
-typeIsVector x =
-    case typeNormalizeForFlatBuf x of
-         TOpaque{..} -> elem typeName sET_TYPES || typeName == mAP_TYPE
-         _ -> False
+typeIsVector x = typeIsIterable ?d $ typeNormalizeForFlatBuf x
```

Comment:
this function name is a bit misleading. Perhaps it should be typeIsSerializedAsVector